e2e/cypress/integration/api/iam/04_teams_api.spec.ts

```diff
@@ -0,0 +1,318 @@
+  };
+  const cypressPrefix = 'test-teams-api';
+  const now = Cypress.moment().format('MMDDYYhhmm');
+  const project1 = {
+      id: `${cypressPrefix}-project1-${now}`,
+      name: 'Test Project 1'
+  };
+  const project2 = {
+      id: `${cypressPrefix}-project2-${now}`,
+      name: 'Test Project 2'
+  };
+
+  before(() => {
+    cy.adminLogin('/').then(() => {
+      const admin = JSON.parse(<string>localStorage.getItem('chef-automate-user'));
+      adminToken = admin.id_token;
+      defaultAdminReq.auth.bearer = adminToken;
+      cy.cleanupV2IAMObjectsByIDPrefixes(adminToken, cypressPrefix,
+        ['policies', 'projects', 'teams']);
+    });
+  });
+
+  after(() => {
+    cy.cleanupV2IAMObjectsByIDPrefixes(adminToken, cypressPrefix,
+      ['policies', 'projects', 'teams']);
+  });
+
+  describeIfIAMV2p1('project assignment enforcement', () => {
+    let nonAdminToken = '';
+    const nonAdminTokenID = `${cypressPrefix}-nonadmin-token-${now}`;
+    const teamID = `${cypressPrefix}-team-${now}`;
+    const defaultNonAdminReq = {
+      headers: { 'api-token': nonAdminToken },
+      method: 'POST',
+      url: '/apis/iam/v2beta/teams'
+    };
+    const scenarios = [
+      {req: defaultNonAdminReq, id: 'project-1-admin'},
+      {req: defaultAdminReq, id: 'superadmin'}
+    ];
+
+    before(() => {
+      for (const project of [project1, project2]) {
+        cy.request({
+            auth: { bearer: adminToken },
+            method: 'POST',
+            url: '/apis/iam/v2beta/projects',
+            body: project
+        });
+      }
+
+      cy.request({ ...defaultAdminReq,
+        method: 'POST',
+        url: '/apis/iam/v2beta/tokens',
+        body: {
+            id: nonAdminTokenID,
+            name: 'Nonadmin Token'
+        }
+      }).then((response) => {
+          nonAdminToken = response.body.token.value;
+          defaultNonAdminReq.headers = { 'api-token': nonAdminToken };
+      });
+
+      cy.request({ ...defaultAdminReq,
+        url: '/apis/iam/v2beta/policies',
+        method: 'POST',
+        body: {
+            name: 'token-access',
+            id: `${cypressPrefix}-token-access-policy-${now}`,
+            members: [ `token:${nonAdminTokenID}` ],
+            statements: [
+              {
+                effect: 'ALLOW',
+                actions: [ 'iam:teams:*' ],
+                projects: [ '*' ]
+              },
+              {
+                effect: 'ALLOW',
+                actions: [ 'iam:projects:assign' ],
+                projects: [ project1.id ]
+              }
+            ],
+            projects: []
+        }
+      });
+    });
+
+    beforeEach(() => {
+      cy.cleanupV2IAMObjectsByIDPrefixes(adminToken, cypressPrefix, ['teams']);
+    });
+
+    afterEach(() => {
+      cy.cleanupV2IAMObjectsByIDPrefixes(adminToken, cypressPrefix, ['teams']);
+    });
+
+    describe('POST /apis/iam/v2beta/teams', () => {
+      context('both superadmin and project1 admin', () => {
```

Comment:
Cypress doesn't seem to differentiate between `describe` and `context`, so could we just use `describe`? 

I'd like if we had less layering in general in our test structure, since the test names tend to get long and hard to parse. Or if the layering seems necessary, let's pay attention to what the test name comes out to and change the content of each layer so it reads clearly. 

for example, here's what we have now:
```
POST /apis/iam/v2beta/teams
     both superadmin and project1 admin
         project-1-admin can create a new team with no projects
         superadmin can create a new team with no projects
```

I would prefer to drop the `both superadmin and project1 admin` context because it doesn't add anything to what's below. Or move the scenario loop inside each it block, resulting in output like this:

```
POST /apis/iam/v2beta/teams
     both superadmin and project1 admin
           can create a new team with no projects
           can create a team with project1
           cannot create a team with a project that doesn't exist
```

Reply:
I like the context myself in general (as a distinction from describe), but I'm def on board with the tests reading more like ^^.